ui/app/components/mount-backend-form.ts

```diff
@@ -74,111 +79,111 @@ export default class MountBackendForm extends Component<Args> {
   }
 
   typeChangeSideEffect(type: string) {
-    if (this.args.mountType !== 'secret') return;
-    if (type === 'pki') {
+    if (this.args.mountType === 'secret') {
       // If type PKI, set max lease to ~10years
-      this.args.mountModel.config.maxLeaseTtl = '3650d';
-    } else {
-      // otherwise reset
-      this.args.mountModel.config.maxLeaseTtl = 0;
+      this.args.mountModel.config.maxLeaseTtl = type === 'pki' ? '3650d' : 0;
     }
   }
 
   checkModelValidity(model: MountModel) {
-    const { isValid, state, invalidFormMessage } = model.validate();
+    const { mountType } = this.args;
+    const { isValid, state, invalidFormMessage, data } =
+      mountType === 'secret' ? model.toJSON() : model.validate();
```

Comment:
is it worth adding a comment that `model.validate()` will be updated when the auth ember data model is migrated? It's fairly obvious, but this is a weird case where two different models are used here and I'm not sure what our test coverages is. Just thinking ahead in case this migration work has to be handed off at some point

Reply:
That's a good point and I wonder if it's worth queuing up the auth methods next to reduce the time where these components have to support both form classes and Ember Data models. Let me think on that a minute but I'm leaning towards going in that direction. If not I will sprinkle in some comments.